### DISCUSSION TITLE
heatmap columns

### DISCUSSION THREAD

**Discussion by anderMartin92** [kernc/backtesting.py#710] [Q&A] [answered]
Sorry that the question is basic but I haven't found an answer for several days.

I want to have a list with the names of the columns of the heatmap (the name of the variables that are optimized).
<img width="504" alt="Captura de Pantalla 2022-08-04 a las 3 29 40" src="https://user-images.githubusercontent.com/62348899/182744669-8f358f3a-ba23-4a97-a0f1-4f9af6c2b513.png">

I have tried everything, for i in params. list(rrrr). params.columns.value and much more and I can't get the list of names.
Thank you very much

**Comment by kernc** [comment, marked as answer]
The encircled area looks like `rrrr.index`.

**Reply by anderMartin92** [reply]
params.index returns 0 and rrrr.index I show it to you in the image. Any more ideas? Thank you so much for everything ;
<img width="380" alt="Captura de Pantalla 2022-08-04 a las 14 43 55" src="https://user-images.githubusercontent.com/62348899/182850799-29ba9547-c8e8-4c87-a10e-e686b8da4e80.png">
)

**Reply by kernc** [reply]
Heatmap is a Series
https://github.com/kernc/backtesting.py/blob/8b83b3541a5f00f3d86047caf1997daa966e6dee/backtesting/backtesting.py#L1342-L1346
with a [MultiIndex](https://pandas.pydata.org/docs/reference/api/pandas.MultiIndex.html) index. See if any of its properties (such as `.names`) resolve to what you are after.

**Reply by anderMartin92** [reply]
`rrrr = rrrr.reset_index()
        
        print(list(rrrr))`

with reset.index i have disarmed the multiindex. thank you for all.
